cmd/kube-proxy/app/server_others.go

```diff
@@ -461,6 +461,18 @@ func getLocalDetector(mode proxyconfigapi.LocalMode, config *proxyconfigapi.Kube
 			break
 		}
 		return proxyutiliptables.NewDetectLocalByCIDR(nodeInfo.Spec.PodCIDR, ipt)
+	case proxyconfigapi.LocalModeInterfaceName:
+		if len(strings.TrimSpace(config.InterfaceName)) == 0 {
+			klog.InfoS("Detect-local-mode set to InterfaceName, but no ifacename is defined")
```

Comment:
"but no interface-name is defined"

But also, you shouldn't log an Info message and fall back, you should just return an error. (The `ClusterCIDR` case has to have fallback because it's the default, but the others shouldn't. Though we probably don't want to belatedly fix the `NodeCIDR` case at this point.)